hw/ip/otbn/dv/model/otbn_model.cc

```diff
@@ -257,23 +257,36 @@ static int start_model(OtbnModel *model, const char *imem_scope,
 
 // Step once in the model. Returns 1 if the model has finished, 0 if not and -1
 // on failure. If gen_trace is true, pass trace entries to the trace checker.
-// If the model has finished, writes otbn.ERR_CODE to *err_code.
-static int step_model(OtbnModel *model, bool gen_trace, uint32_t *err_code) {
+// If the model has finished, writes otbn.ERR_BITS to *err_bits.
+static int step_model(OtbnModel *model, bool gen_trace, uint32_t *err_bits) {
   assert(model);
-  assert(err_code);
+  assert(err_bits);
 
   if (!model->ensure())
     return -1;
   assert(model->iss);
   ISSWrapper &iss = *model->iss;
 
   try {
-    std::pair<bool, uint32_t> ret = iss.step(gen_trace);
-    if (ret.first) {
-      *err_code = ret.second;
-      return 1;
-    } else {
-      return 0;
+    std::pair<int, uint32_t> ret = iss.step(gen_trace);
+    switch (ret.first) {
+      case -1:
+        // Something went wrong, such as a trace mismatch. We've already printed
+        // a message to stderr so can just return -1.
+        return -1;
+
+      case 1:
+        // The simulation has stopped. Extract err_bits
+        *err_bits = ret.second;
```

Comment:
We use the err_bits second argument only here? It looks like we could simplify the code by making iss.step() only return the first argument in the pair, and reading ERR_BITS in here if necessary.

Reply:
Good point! That's much nicer.